src/main/java/com/amazon/ion/impl/IonReaderTextSystemX.java

```diff
@@ -191,8 +191,11 @@ private static boolean valueWithinBounds(String value, int len, char[] minImage,
         private static boolean magnitudeLessThanOrEqualTo(String lhs, int lhsLen, char[] rhs)
         {
             assert lhsLen == rhs.length;
-            for (int i = lhsLen - 1; i >= 0; i--)
+            for (int i = 0; i < lhsLen; i++)
             {
+                if (lhs.charAt(i) < rhs[i]) {
+                    return true;
+                }
                 if (lhs.charAt(i) > rhs[i])
                 {
                     return false;
```

Comment:
```suggestion
                char lhc = lhs.charAt(i);
                char rhc = rhs[i];
                if (lhc != rhc) {
                    return lhc < rhc;
```

I'm not super sold on this one, but might as well use the accessors only once at least.